devtools_m1/cypress/integration/Wishlists/steps.js

```diff
@@ -0,0 +1,98 @@
+      cy.get('#product_addtocart_form input[name="super_group[2]"]').clear().type('1')
+      cy.get('#product_addtocart_form input[name="super_group[3]"]').clear().type('1')
+      break;
+    case 'simple':
+    case 'bundle': // For now, we only have one option for each bundle option, so we don't have to do anything.
+      // Do nothing
+      break;
+    default:
+      throw 'Methinks thou hast forgotten something…'
+  }
+  cy.contains('Add to Wishlist').click()
+})
+
+Then('A wishlist {string} event should be sent to Drip', function(type) {
+  cy.log('Validating that the wishlist call has everything we need')
+  cy.wrap(Mockclient.retrieveRecordedRequests({
+    'path': '/v2/123456/events'
+  })).then(function(recordedRequests) {
+    expect(recordedRequests).to.have.length.of.at.least(1)
+    let expectedAction = (type === 'add' ? 'Added item to wishlist' : 'Removed item from wishlist')
+    const wishlistEvent = recordedRequests.find(request => JSON.parse(request.body.string).events[0].action === expectedAction)
+    expect(wishlistEvent).to.not.be.undefined
+
+    const event = JSON.parse(wishlistEvent.body.string).events[0]
+    expect(event.email).to.eq('testuser@wishlist.example.com')
+    expect(event.properties.product_id).to.eq('1')
+    expect(event.properties.categories).to.eq('')
+    expect(event.properties.brand).to.eq(false)
+    expect(event.properties.name).to.eq('Widget 1')
+    expect(event.properties.price).to.eq(1122)
+    expect(event.properties.currency).to.eq('USD')
+    expect(event.properties.image_url).to.eq('http://main.magento.localhost:3005/media/catalog/product/')
+    expect(event.properties.source).to.eq('magento')
+  })
+})
+
+When('I remove the {string} widget from my wishlist via {string}', function(type, method) {
+  // For some reason, Magento throws an error here in JS. We don't really care, so ignore it.
+  cy.on('uncaught:exception', (err, runnable) => {
+    return false
+  })
+  cy.visit(`${getCurrentFrontendDomain()}/wishlist/index/index/`)
+  switch (type) {
+    case 'configurable':
+      cy.get('#product-options-wrapper select').select('XL')
+      break;
+    case 'grouped':
+      cy.get('#product_addtocart_form input[name="super_group[2]"]').clear().type('1')
+      cy.get('#product_addtocart_form input[name="super_group[3]"]').clear().type('1')
+      break;
+    case 'simple':
+    case 'bundle': // For now, we only have one option for each bundle option, so we don't have to do anything.
+      // Do nothing
+      break;
+    default:
+      throw 'Methinks thou hast forgotten something…'
+  }
+  switch(method) {
+    case 'quantity':
+      cy.get('input[class="input-text qty validate-not-negative-number"]').type('{selectall}0')
+      cy.get('button[title="Update Wishlist"]').first().click()
+      break;
+    case 'the trashcan':
+      cy.get('a[title="Remove Item"]').click()
+      break;
+    default:
+      throw 'Methinks thou hast forgotten something…'
+  }
+})
```

Comment:
Nit: Could we have a newline at the end of the file?

Reply:
Damn you VSCode for not maintaining my IntelliJ setting.